stock_request_picking_type/models/stock_request_order.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2019 Open Source Integrators
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class StockRequestOrder(models.Model):
+    _inherit = "stock.request.order"
+
+    @api.model
+    def _get_default_picking_type(self):
+        companies = self._context.get("allowed_company_ids", []).copy()
+        companies.append(False)
+        return (
+            self.env["stock.picking.type"]
+            .search(
+                [
+                    ("code", "=", "stock_request_order"),
+                    "|",
+                    ("warehouse_id.company_id", "in", companies),
```

Comment:
Moreover, I think this is not necessary as there is a record rule on picking type. I'm wondering why it is not sufficent to search on...